src/Relation/Binary/Construct/Closure/Reflexive/Properties/WithK.agda

```diff
@@ -0,0 +1,59 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of reflexive closures which rely on the K rule
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe #-}
+
+module Relation.Binary.Construct.Closure.Reflexive.Properties.WithK where
+
+open import Data.Empty.Irrelevant using (⊥-elim)
+open import Data.Sum as Sum
+open import Relation.Binary
+open import Relation.Binary.Construct.Closure.Reflexive
+open import Relation.Binary.Construct.Closure.Reflexive.Properties public
+import Relation.Binary.PropositionalEquality as PropEq
+open PropEq using (_≡_; refl)
+
+module _ {a ℓ} {A : Set a} {_~_ : Rel A ℓ} where
+    irrel∧irrefl⟶irrel : Irrelevant _~_ → Irreflexive _≡_ _~_ → Irrelevant (Refl _~_)
+    irrel∧irrefl⟶irrel ~-irrel _ [ x∼y ] [ x∼y′ ] = PropEq.cong [_] (~-irrel x∼y x∼y′)
+    irrel∧irrefl⟶irrel _ ~-irrefl [ x∼y ] refl = ⊥-elim (~-irrefl refl x∼y)
+    irrel∧irrefl⟶irrel _ ~-irrefl refl [ x∼y ] = ⊥-elim (~-irrefl refl x∼y)
+    irrel∧irrefl⟶irrel _ _ refl refl = refl
+
+    asym⟶antisym : ∀ {ℓ'} {_≈_ : Rel _ ℓ'} → Reflexive _≈_ → Asymmetric _~_ → Antisymmetric _≈_ (Refl _~_)
+    asym⟶antisym ≈-refl ~-asym [ x∼y ] [ x∼y′ ] = ⊥-elim (~-asym x∼y x∼y′)
+    asym⟶antisym ≈-refl _ _ refl = ≈-refl
+    asym⟶antisym ≈-refl _ refl _ = ≈-refl
+
+    partialOrder : IsStrictPartialOrder _≡_ _~_ → IsPartialOrder _≡_ (Refl _~_)
+    partialOrder ~-IsStrictPartialOrder = record
+        { isPreorder = record
+            { isEquivalence = isEquivalence
+            ; reflexive = λ { refl → refl }
+            ; trans = trans (IsStrictPartialOrder.trans ~-IsStrictPartialOrder) }
+        ; antisym = asym⟶antisym {_} {_≡_} Eq.refl asym }
+      where open IsStrictPartialOrder ~-IsStrictPartialOrder hiding (trans)
+
+    decPartialOrder : IsDecStrictPartialOrder _≡_ _~_ → IsDecPartialOrder _≡_ (Refl _~_)
+    decPartialOrder ~-IsDecStrictPartialOrder = record
+      { isPartialOrder = partialOrder isStrictPartialOrder; _≟_ = _≟_; _≤?_ = dec _≟_ _<?_ }
+      where open IsDecStrictPartialOrder ~-IsDecStrictPartialOrder
+
+    totalOrder : IsStrictTotalOrder _≡_ _~_ → IsTotalOrder _≡_ (Refl _~_)
+    totalOrder ~-IsStrictTotalOrder = record
+      { isPartialOrder = partialOrder isStrictPartialOrder; total = total }
+      where open IsStrictTotalOrder ~-IsStrictTotalOrder
+
+            total : Total (Refl _~_)
```

Comment:
This feels like it should go in its own top-level proof

Reply:
I think `total` requires `IsStrictTotalOrder` anyhow, so it's not really a win — one can rather merely use `IsTotalOrder.total (isTotalOrder ?)`.

Reply:
Looks to me like it only requires `Trichotomous`? You can use `IsTotalOrder.total (isTotalOrder ?)` but it's considerably longer than `total`.